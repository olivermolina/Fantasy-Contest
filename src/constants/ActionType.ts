export enum ActionType {
  /*
   *  Deposit cash balance
   */
  PAY = 'PAY',
  /*
   *  Withdraw cash balance
   */
  PAYOUT = 'PAYOUT',
  /*
   *  Customer signup
   */
  CREATE_CUSTOMER = 'CREATE_CUSTOMER',
  /*
   *  GIDX user profile API query
   */
  GET_CUSTOMER_PROFILE = 'GET_CUSTOMER_PROFILE',
  /*
   *  Adding new payment method
   */
  SAVE_PAYMENT_METHOD = 'SAVE_PAYMENT_METHOD',
  /*
   *  GIDX payment details API query
   */
  PAYMENT_DETAILS = 'PAYMENT_DETAILS',
  /*
   *  Add bet/entry
   */
  PLACE_BET = 'PLACE_BET',
  /*
   *  Joining a contest
   */
  JOIN_CONTEST = 'JOIN_CONTEST',
  /*
   *  GIDX customer monitor API query
   */
  CUSTOMER_MONITOR = 'CUSTOMER_MONITOR',
  /*
   *  Winning a contest
   */
  TOKEN_CONTEST_WIN = 'TOKEN_CONTEST_WIN',
  /*
   *  Winning a cash bet/entry
   */
  CASH_CONTEST_WIN = 'CASH_CONTEST_WIN',
  /*
   *  Adding free credits to a user
   */
  ADD_FREE_CREDIT = 'ADD_FREE_CREDIT',
  /*
   *  GIDX customer registration API query
   */
  CUSTOMER_REGISTRATION = 'CUSTOMER_REGISTRATION',
  /*
   *  Refund bet/entry for no action picks
   */
  CASH_CONTEST_CANCELLED = 'CASH_CONTEST_CANCELLED',
  /*
   * Increment user's amount available to withdraw
   */
  ADD_WITHDRAWABLE = 'ADD_WITHDRAWABLE',
  /*
   * Decrease user's amount available to withdraw
   */
  REMOVE_WITHDRAWABLE = 'REMOVE_WITHDRAWABLE',
  /*
   * Convert cash balance to bonus credit
   * when the user accepts the offer to keep the cash balance on the platform
   */
  WITHDRAW_BONUS_CREDIT = 'WITHDRAW_BONUS_CREDIT',
}
