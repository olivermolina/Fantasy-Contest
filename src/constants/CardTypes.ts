export enum CardTypes {
  amex = 'american-express',
  visa = 'visa',
  discover = 'discover',
  mastercard = 'mastercard',
}
