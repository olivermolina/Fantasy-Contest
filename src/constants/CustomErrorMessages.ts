export enum CustomErrorMessages {
  SIGNUP_ERROR = 'There was an error registering. Please try again later.',
  SIGNUP_DUPLICATE_ERROR = 'Sorry, user already registered.',
  NOT_VERIFIED = 'Your account is not verified. Please contact support with questions!',
  FAILED_VERIFICATION = 'User verification failed. You have provided incorrect/invalid information. Please try again.',
  GIDX_DEFAULT = 'Oops! Something went wrong with your account. Please contact support with questions!',
  'ID-UA18' = 'Looks like you’re not of age to participate on our site. Please contact support with questions!',
  'ID-UA19' = 'Looks like you’re not of age to participate on our site. Please contact support with questions!',
  'ID-UA21' = 'Looks like you’re not of age to participate on our site. Please contact support with questions!',
  'ID-EX' = 'Whoops looks like you already have an account in our system. Please login to your original account.',
  'ID-BLOCK' = 'Your account isn’t active please contact support with questions.',
  'ID-HR' = 'Your account isn’t active please contact support with questions.',
  'ID-HVEL-ACTV' = 'Please contact support to verify your account.',
  'LL-BLOCK' = `Unfortunately you're in a location we aren't yet live in. Please check back soon as we always are adding more states! Contact us with Q's.`,
  BLOCKED_COLLEGE_SPORTS_ERROR = 'Sorry, your state doesn’t allow contests based on college sports.',
  ACCOUNT_TRANSACTION = 'Sorry, we were unable to process your transaction at this time. Please contact support with questions!',
  PAYMENT_METHOD = 'Sorry, we were unable to save your payment method at this time. Please contact support with questions!',
  CONTACT_US = 'Sorry, something went wrong. Please try again later!',
  LOGIN = 'Sorry, we can’t sign in to your account.  Please contact support with questions!',
  PASSWORD_RESET = 'There was an error sending password reset link.',
  PASSWORD_RESET_EXPIRED = 'Password reset token has expired/invalid. Please try resetting your password via email again.',
  PASSWORD_UPDATE_LINK = 'There was an error updating your password. Please request a new password reset link.',
  SUBMIT_ENTRY_ERROR = 'Error submitting entry. Please try again!',
  INSUFFICIENT_FUNDS_ERROR = 'Insufficient funds. Please deposit funds in your account and try again.',
}
