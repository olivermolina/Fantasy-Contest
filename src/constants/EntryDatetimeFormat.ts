export const EntryDatetimeFormat = 'MMM DD, YYYY hh:mm A';
export const EntryDateFormat = 'MMM DD, YYYY';
