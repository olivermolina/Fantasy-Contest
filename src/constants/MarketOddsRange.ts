export enum MarketOddsRange {
  MIN = -150,
  MAX = 150,
}
