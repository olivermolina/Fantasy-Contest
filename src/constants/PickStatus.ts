export enum PickStatus {
  PENDING = 'PENDING',
  SETTLED = 'SETTLED',
  WIN = 'WIN',
  LOSS = 'LOSS',
}
