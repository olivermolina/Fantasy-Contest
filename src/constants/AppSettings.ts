import { ReloadBonusType } from '~/constants/ReloadBonusType';

export const DefaultAppSettings = {
  MIN_BET_AMOUNT: 1,
  MAX_BET_AMOUNT: 100,
  REPEAT_ENTRIES: 0,
  MAX_RETENTION_BONUS: 100,
  RETENTION_BONUS_MATCH_MULTIPLIER: 1,
  RETENTION_BONUS_WEEKLY_CHANCE: 1,
  NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
  STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
  BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: 25,
  REFERRAL_CREDIT_AMOUNT: 25,
  MAX_MATCH_DEPOSIT_AMOUNT: 100,
  REFERRAL_CUSTOM_TEXT:
    'Refer a friend from the link or code in your profile and get $25 in bonus credit!',
  DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
  RELOAD_BONUS_TYPE: ReloadBonusType.FLAT,
  RELOAD_BONUS_AMOUNT: 100,
  MIN_MARKET_ODDS: -150,
  MAX_MARKET_ODDS: 150,
  CHALLENGE_PROMO_MESSAGE:
    'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
  SIGNUP_FREE_ENTRY: 1,
  MAX_DAILY_TOTAL_BET_AMOUNT: 0,
  WEEKLY_REFERRAL_MAX_AMOUNT_EARNED: '0',
  HOMEPAGE_HEADING_1: 'Daily Fantasy Contests!',
  HOMEPAGE_HEADING_2:
    'Select More or Less on player stats to win up to 35x your cash!',
  ACH_DEPOSIT_AUTO_APPROVE_COUNT: 3,
  MIN_DEPOSIT_AMOUNT: 20,
};
