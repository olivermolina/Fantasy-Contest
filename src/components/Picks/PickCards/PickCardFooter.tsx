import React from 'react';
import { PickStatus } from '~/constants/PickStatus';
import { CheckCircle, XCircle } from '~/components/Picks/Icons';
import { BetStakeType } from '@prisma/client';
import { Divider, Stack } from '@mui/material';
import { grey } from '@mui/material/colors';
import { InsuredPayoutInterface } from '~/components/Picks/PickCards/StraightCard';

interface Props {
  risk?: number;
  status?: PickStatus;
  potentialWin?: number | InsuredPayoutInterface;
  stakeType?: BetStakeType;
}

const PickCardFooter = (props: Props) => {
  const pickStatus = props.status;

  return (
    <div className="p-1 md:p-5 flex flex-row justify-between">
      <div className={'flex flex-col items-center'}>
        <p className="text-gray-400 text-sm">Risk</p>
        <p className="font-bold">{props.risk}</p>
      </div>
      <div className={'flex flex-col items-center'}>
        <p className="text-gray-400 text-sm">
          {pickStatus === PickStatus.WON ? 'Winnings' : 'Potential Win'}
        </p>
        {props.stakeType === BetStakeType.ALL_IN ? (
          <p className="font-bold">{props.potentialWin as number}</p>
        ) : (
          <Stack
            sx={{
              border: 1,
              borderRadius: 1,
              borderColor: grey[200],
              flexGrow: 1,
            }}
            direction={'row'}
            justifyContent={'space-evenly'}
            alignItems="center"
            spacing={1}
            className={'text-sm'}
          >
            <div
              className={
                'flex flex-col justify-center items-center p-1 text-gray-500'
              }
            >
              <span>Correct</span>
              <span className={'text-bold font-bold text-black'}>
                {(props.potentialWin as InsuredPayoutInterface).numberOfPicks -
                  1}
                /{(props.potentialWin as InsuredPayoutInterface).numberOfPicks}
              </span>
            </div>
            <div
              className={
                'flex flex-col justify-center items-center p-1 text-gray-500'
              }
            >
              <span>Payout</span>
              <span className={'text-bold font-bold text-black'}>
                $
                {
                  (props.potentialWin as InsuredPayoutInterface)
                    .secondaryInsuredPayout
                }
              </span>
            </div>
            <Divider orientation="vertical" flexItem />
            <div
              className={
                'flex flex-col justify-center items-center p-1 text-gray-500'
              }
            >
              <span>Correct</span>
              <span className={'text-bold font-bold text-black'}>
                {(props.potentialWin as InsuredPayoutInterface).numberOfPicks}/
                {(props.potentialWin as InsuredPayoutInterface).numberOfPicks}
              </span>
            </div>
            <div
              className={
                'flex flex-col justify-center items-center p-1 text-gray-500'
              }
            >
              <span>Payout</span>
              <span className={'text-bold font-bold text-black'}>
                $
                {
                  (props.potentialWin as InsuredPayoutInterface)
                    .primaryInsuredPayout
                }
              </span>
            </div>
          </Stack>
        )}
      </div>
      <div className={'flex flex-col items-center'}>
        <p className="text-gray-400 text-sm">Status</p>
        <div className={'flex flex-row gap-1 items-center'}>
          {pickStatus === PickStatus.PENDING ? (
            <div className="w-4 h-4 border border-gray-300 rounded-full" />
          ) : null}
          {pickStatus === PickStatus.LOST ? (
            <XCircle className="w-6 h-6 fill-red-500" />
          ) : null}

          {pickStatus === PickStatus.WON ? (
            <CheckCircle className="w-6 h-6 fill-green-500" />
          ) : null}
          <p className="font-bold capitalize text-sm">{pickStatus}</p>
        </div>
      </div>
    </div>
  );
};

export default PickCardFooter;
