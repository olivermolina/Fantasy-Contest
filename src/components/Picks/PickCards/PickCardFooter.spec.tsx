import React from 'react';
import { render, screen } from '@testing-library/react';
import PickCardFooter from './PickCardFooter';
import { PickStatus } from '~/constants/PickStatus';
import { BetStakeType } from '@prisma/client';

describe('PickCardFooter', () => {
  it('should render', () => {
    const props = {
      risk: 10,
      status: PickStatus.PENDING,
      potentialWin: 100,
      stakeType: BetStakeType.ALL_IN,
      payout: 100,
      bonusCreditStake: 10,
    };
    const { container } = render(<PickCardFooter {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should not show bonus credit column if it is not admin view', () => {
    const props = {
      risk: 10,
      status: PickStatus.PENDING,
      potentialWin: 100,
      stakeType: BetStakeType.ALL_IN,
      payout: 100,
      bonusCreditStake: 10,
      isAdminView: false,
    };
    render(<PickCardFooter {...props} />);

    expect(screen.queryAllByText('Bonus Credit')).toHaveLength(0);
  });

  it('should show bonus credit column if it is admin view', () => {
    const props = {
      risk: 10,
      status: PickStatus.PENDING,
      potentialWin: 100,
      stakeType: BetStakeType.ALL_IN,
      payout: 100,
      bonusCreditStake: 10,
      isAdminView: true,
    };
    render(<PickCardFooter {...props} />);

    expect(screen.queryAllByText('Bonus Credit')).toHaveLength(1);
  });
});
