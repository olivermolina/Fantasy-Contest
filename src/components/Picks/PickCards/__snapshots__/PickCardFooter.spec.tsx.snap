// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickCardFooter should render 1`] = `
<div>
  <div
    class="p-1 md:p-2 flex flex-row justify-between gap-1 text-white"
  >
    <div
      class="flex flex-col items-center"
    >
      <p
        class="text-lightText text-sm"
      >
        Entry
      </p>
      <div
        class="flex font-bold h-full items-center"
      >
        $10.00
      </div>
    </div>
    <div
      class="flex flex-col items-center"
    >
      <p
        class="text-lightText text-sm"
      >
        Potential Win
      </p>
      <div
        class="font-bold flex font-bold h-full items-center"
      >
        $100.00
      </div>
    </div>
    <div
      class="flex flex-col items-center"
    >
      <p
        class="text-lightText text-sm"
      >
        Status
      </p>
      <div
        class="flex flex-col md:flex-row gap-1 items-center h-full"
      >
        <div
          class="w-4 h-4 border border-gray-300 rounded-full"
        />
        <p
          class="font-bold capitalize text-sm"
        >
          PENDING
        </p>
      </div>
    </div>
  </div>
</div>
`;
